### PR TITLE
Roles and Perms Matrix

### PR DESCRIPTION
**Pull request by Siboney1**
Breaks out the Perm & Roles

### CODE REVIEW

**Comment by spectro-prow** [comment]
Hi @Siboney1. Thanks for your PR.

I'm waiting for a [spectrocloud](https://github.com/orgs/spectrocloud/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/spectrocloud/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by Siboney1** [comment]
First Draft

File: content/docs/08-user-management/4-roles-permissions-and-project-scope.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+</Tabs.TabPane>
+
+<Tabs.TabPane tab="Editor" key="Project Editor">
+<br />
+
+## Project Editor
+
+|                    |         |          |            | **Permissions** |            |            |             |             |            |
+| :----------------- | :-----: | :------: | :--------: | :-------------: | :--------: | :--------: | :---------: | :---------: | :--------: |
+|                    | **Get** | **List** | **Delete** |   **Backup**    | **Import** | **Create** | **Publish** | **Restore** | **Update** |
```
Comment (resmaa) [inline content/docs/08-user-management/4-roles-permissions-and-project-scope.md:83]: Plz move the "update" column after  "List" column

File: content/docs/08-user-management/4-roles-permissions-and-project-scope.md [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+<br />
+
+<Tabs>
+
+<Tabs.TabPane tab="Admin" key="Project Admin">
+<br />
+
+## Project Administrator
+
+|                    |         |          |            | **Permissions** |            |            |             |             |            |
```
Comment (resmaa) [inline content/docs/08-user-management/4-roles-permissions-and-project-scope.md:56]: image

First Column can be named "Permissions" and all other columns can be give "operations"

File: content/docs/08-user-management/4-roles-permissions-and-project-scope.md [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+
+</Tabs.TabPane>
+
+<Tabs.TabPane tab="Viewer" key="Project Viewer">
+
+<br />
+
+## Project Viewer
+
+|                    |         |          |            | **Permissions** |            |            |             |             |            |
```
Comment (resmaa) [inline content/docs/08-user-management/4-roles-permissions-and-project-scope.md:112]: Column headings to be updated as proposed above

**Comment by resmaa** [comment]
in doc lets have  tenant scope first and then project scope.
image

**Review by prathabk — COMMENTED** [review]
@Siboney1 Overall looks good. A suggestion on the permission/role matrix. The matrix table looks verbose like some permission like import, publish, backup, restore is applicable only to one resource which occupies the whole column. Hence, we can simply string like cluster.import. For reference take a look at the GCP docs https://cloud.google.com/iam/docs/understanding-roles
image

**Comment by Siboney1** [comment]
> @Siboney1 Overall looks good. A suggestion on the permission/role matrix. The matrix table looks verbose like some permission like import, publish, backup, restore is applicable only to one resource which occupies the whole column. Hence, we can simply string like cluster.import. For reference take a look at the GCP docs https://cloud.google.com/iam/docs/understanding-roles
> image


I see what you mean. While designing this matrix, I took into consideration that it is a long list (close to 500 entries). I wanted it to be user friendly, minimizing the long scrolls and leveraging tabs so that it looks like a snapshot. Using the checkmarks and including all the possible perms as columns, even though the rest of the columns are blank, makes it look clean.  See what I had before:

Screen Shot 2022-06-09 at 10 06 41 PM

**Comment by resmaa** [comment]
Lets remove the duplicate files
image

**Review by Siboney1 — COMMENTED** [review]
Swapped pages

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/librarium/pull/724#pullrequestreview-1001801755" title="Approved">resmaa</a>*, *<a href="https://github.com/spectrocloud/librarium/pull/724#" title="Author self-approved">Siboney1</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign 
You can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Flibrarium).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/librarium/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**